### PR TITLE
RAIL-4356 - Cannot drill at empty value with JS error

### PR DESCRIPTION
**Pull request by hackerstanislav**
JIRA: RAIL-4356

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8783/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10627/ : SUCCESS in 15m 24s

File: libs/sdk-ui/src/base/vis/drilling.ts [hunk 1]
```diff
@@ -128,3 +129,18 @@ export function fireDrillEvent(
...
+    const { attributeHeaderItem } = attributeItem;
+
+    // This is special behaviour for tiger when we allowed empty string or null
+    // In this case if uri is one of these special value, set on same value also name
+    // because there is now some special ui value (from example "(empty value)") which is not
+    // valid for this case
+    const values = ["", null] as Array<any>;
+    const isEmpty = values.indexOf(attributeHeaderItem.uri) >= 0;
+    // Send empty string for not, need to be updated for NULL in future
+    attributeHeaderItem.name = isEmpty ? "" : attributeHeaderItem.name;
```
Comment (no23reason) [inline libs/sdk-ui/src/base/vis/drilling.ts:143]: This changes the input, is this desired? Should we instead return a new object?

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2744

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2516/ : SUCCESS in 1m 17s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2815/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1640/ : SUCCESS in 20m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10640/ : SUCCESS in 16m 47s

**Pull request merged**